python/lsst/ts/mtreflector/controller.py

```diff
@@ -0,0 +1,437 @@
+# This file is part of ts_mtreflector.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["LabjackChannel", "MTReflectorController"]
+import asyncio
+import functools
+import logging
+import types
+from typing import Any
+
+import yaml
+
+# Hide mypy error `Module "labjack" has no attribute "ljm"`.
+from labjack import ljm  # type: ignore
+from lsst.ts import utils
+from lsst.ts.ess.labjack import BaseLabJackDataClient
+from lsst.ts.xml.enums.MTReflector import MTReflectorStatus
+
+# Time limit for communicating with the LabJack (seconds).
+COMMUNICATION_TIMEOUT = 5
+# Sleep time before trying to reconnect (seconds).
+RECONNECT_WAIT = 60
+
+
+class LabjackChannel:
```

Comment:
I don't like the implementation and use of this class. In ts_ess_labjack a similar challenge was faced and it was solved by including the channel names, offsets and scales in the config instead of deriving them here. Please adopt that solution.